### PR TITLE
Newrelic

### PR DESCRIPTION
**Pull request by ametaireau**
Based on #187

### CODE REVIEW

File: cliquet/__init__.py [hunk 1]
```diff
@@ -327,3 +340,16 @@ def initialize_cliquet(config, version=None, project_name=None):
     # Include cliquet views with the correct api version prefix.
     config.include("cliquet", route_prefix=api_version)
     config.route_prefix = api_version
+
+
+def wrap_app(app, settings):
```
Comment (Natim) [inline cliquet/__init__.py]: I don't like too much the name of this method, should we name it ``plug_newrelic_if_activated``

File: cliquet/__init__.py [hunk 2]
```diff
@@ -327,3 +340,16 @@ def initialize_cliquet(config, version=None, project_name=None):
...
+    "Wraps the passed wsgi application with wsgi middlewares."
+
+    # Setup new-relic.
+    if settings.get('cliquet.newrelic_config', False):
+        ini_file = settings.get('cliquet.newrelic_config')
+        env = settings.get('cliquet.newrelic_env', 'dev')
```
Comment (leplatrem) [inline cliquet/__init__.py]: Don't we want default value for this setting ? Even None ?

File: cliquet/__init__.py (hunk 1 above)
Comment (leplatrem) [inline cliquet/__init__.py]: Yes, maybe something like ``wrap_newrelic_app`` ?

File: cliquet/__init__.py (hunk 2 above)
Comment (Natim) [inline cliquet/__init__.py]: Here the default is ``dev`` isn't it enough?

File: cliquet/__init__.py (hunk 1 above)
Comment (ametaireau) [inline cliquet/__init__.py]: The idea is to have this wrap other things, not only new relic, and have the wrappers defined in a configuration, so we need an app wrapper for all the middlewares.

File: cliquet/__init__.py (hunk 2 above)
Comment (ametaireau) [inline cliquet/__init__.py]: We could have a default value for `'cliquet.newrelic_config` set to None, yes.

File: cliquet/tests/test_initialization.py [hunk 3]
```diff
@@ -85,6 +85,67 @@ def test_initialize_cliquet_is_deprecated(self):
...
+
+    @mock.patch('cliquet.newrelic.agent')
+    def test_newrelic_is_not_included_by_default(self, mocked_newrelic):
+        settings = {}
+        app = cliquet.install_middlewares(mock.sentinel.app, settings)
+        mocked_newrelic.initialize.assert_not_called()
+        self.assertEquals(app, mock.sentinel.app)
+
+
+class SentryConfigurationTest(unittest.TestCase):
```
Comment (Natim) [inline cliquet/tests/test_initialization.py]: Why is this present here? It has been removed in master.

File: cliquet/__init__.py (hunk 2 above)
Comment (leplatrem) [inline cliquet/__init__.py]: Well, I believe we should keep up with gathering all default settings in one place. It's also a very easy way to get a list of all settings handled by cliquet :)

Comment (ametaireau) [inline cliquet/__init__.py]: Yes, dunno why I haven't done that. PR updated.

File: docs/quickstart.rst [hunk 4]
```diff
@@ -35,8 +35,8 @@ just add some extra initialization code:
         config = Configurator(settings=settings)
 
         cliquet.initialize(config, __version__)
-
-        return config.make_wsgi_app()
+        app = config.make_wsgi_app()
+        return cliquet.install_middlewares(app)
```
Comment (leplatrem) [inline docs/quickstart.rst]: Question: don't we want to leave the wrapping only in the monitoring section ?

I fear one thing : the first impression of cliquet might be : "this is too much initialization magic", "too far from vanilla pyramid"

If one day we have general purpose middleware we will add it here, in the general introduction docs.

Thoughts ?

Comment (ametaireau) [inline docs/quickstart.rst]: makes sense.

**Comment by ametaireau** [comment]
updated.

**Pull request merged**